### PR TITLE
Update service graph docs

### LINKED ISSUE #42448: Docs: Add documentation for Tempo service graph feature
Add docs for the service graph feature in Tempo, see PR https://github.com/grafana/grafana/pull/37661.

For similar existing docs see service map section in https://grafana.com/grafana/plugins/grafana-x-ray-datasource/

### PR DESCRIPTION
**Pull request by connorlindsey**
**What this PR does / why we need it**:
Adds service graph info to the Tempo docs. Adds info on some recently added config as well.

**Which issue(s) this PR fixes**:

Fixes #42448

**Special notes for your reviewer**:

### CODE REVIEW

**Review by aocenas — APPROVED** [review]
Looks good to me, would be nice to have a link to Tempo setup for this.

File: docs/sources/datasources/tempo.md [hunk 1]
```diff
@@ -103,6 +115,33 @@ Here is an example JSON:
 }
 ```
 
+## Service Graph
+
+Service Graph in Grafana enables customers to view the graph of their application's architecture. Each node on the graph represents a service such as an API or database. With this graph, customers can easily detect performance issues, increases in error, fault, or throttle rates in any of their services, and dive deep into corresponding traces and root cause.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/tempo.md:120]:
```suggestion
Service Graph in Grafana enables customers to view the graph of their application's architecture. Each node on the graph represents a service such as an API or database. With this graph, customers can easily detect performance issues, increases in error, fault, or throttle rates in any of their services, and dive deep into corresponding traces and root causes.
```

**Review by achatterjee-grafana — COMMENTED** [review]
Suggested a minor grammatical fix.

File: docs/sources/datasources/tempo.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (connorlindsey) [reply docs/sources/datasources/tempo.md]: Added a link to the agent config

**Pull request merged**